### PR TITLE
feat(l10n): Add support for msgctxt when translating.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Use a triple / (///) to give a `t` a context that will
be used as a comment for the l10n team.

issue #3128

### CODE REVIEW

**Comment by vladikoff** [comment]
Added the forked repo that generates:

```
#: app/scripts/views/sign_in.js:91
#. button text
msgctxt "button text"
msgid "Sign in"
msgstr ""

#: app/scripts/views/sign_in.js:94
#. header text
msgctxt "header text"
msgid "Sign in"
msgstr ""
``` 

Need to test if this maps correctly though

File: app/scripts/lib/translator.js [hunk 1]
```diff
@@ -44,12 +44,16 @@ define(function (require, exports, module) {
...
      * @returns {String}
      */
-    get (key, context) {
-      var translation = this.__translations__[key];
+    get (stringToTranslate, context = {}) {
+      let key = stringToTranslate;
+      if (context.msgctxt) {
+        key = context.msgctxt + '\u0004' + stringToTranslate;
+      }
+      var translation = this.__translations__[key] || this.__translations__[stringToTranslate];
```
Comment (vladikoff) [inline app/scripts/lib/translator.js]: let's add "Sign in" to strings.js

File: app/scripts/views/sign_in.js [hunk 2]
```diff
@@ -86,10 +86,18 @@ define(function (require, exports, module) {
       var hasSuggestedAccount = suggestedAccount.get('email');
       var email = this.getEmail();
 
+      /// button text
```
Comment (vladikoff) [inline app/scripts/views/sign_in.js]: should we rename this?

**Comment by shane-tomlinson** [comment]
@vladikoff - updated!

**Comment by vladikoff** [comment]
after an email thread, this seems ready to go

**Pull request merged**